File: Onboarding/app/src/main/java/com/example/onboarding/SplashFragment.java
package com.example.onboarding;

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.google.android.material.tabs.TabLayoutMediator

class SplashFragment : Fragment() {

private var _binding: SplashFragmentBinding? = null
private val binding get() = _binding!!

        override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
        ): View {
        _binding = SplashFragmentBinding.inflate(inflater, container, false)
        return binding.root
        }

        override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        //1
        Handler(Looper.getMainLooper()).postDelayed({
        if (isOnBoardingFinished()) {
        findNavController().navigate(R.id.action_splashFragment_to_homeFragment)
        } else {
        findNavController().navigate(R.id.action_splashFragment_to_onBoardingFragment)
        }
        }, 1500)
        }

//2
private fun isOnBoardingFinished(): Boolean {
        val prefs = requireActivity().getSharedPreferences("onBoarding", Context.MODE_PRIVATE)
        return prefs.getBoolean("finished", false)
        }

        override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
        }

        }

File: Onboarding2/app/src/main/java/com/example/onboarding2/SplashFragment.kt
package com.example.onboarding2

import android.content.Context
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.navigation.fragment.findNavController
import com.example.onboarding2.databinding.SplashFragmentBinding

class SplashFragment : Fragment() {

    private var _binding: SplashFragmentBinding? = null
    private val binding get() = _binding!!

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _binding = SplashFragmentBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        //Splash화면은 Hanlder()를 사용하여 1.5초의 딜레이를 주었습니다. isOnBoardingFinished() 메소드를 호출하여 최초실행여부를 판단해줍니다.
        Handler(Looper.getMainLooper()).postDelayed({
            if (isOnBoardingFinished()) {
                findNavController().navigate(R.id.action_splashFragment_to_homeFragment)
            } else {
                findNavController().navigate(R.id.action_splashFragment_to_onBoardingFragment)
            }
        }, 1500)
    }

    //isOnBoardingFinished() 메소드는 SharedPreferences에 저장된 최초 실행여부를 불러오기 합니다.
    private fun isOnBoardingFinished(): Boolean {
        val prefs = requireActivity().getSharedPreferences("onBoarding", Context.MODE_PRIVATE)
        return prefs.getBoolean("finished", false)
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }

}